keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -565,53 +592,31 @@ class SqlArtifactRepository(
       return false
     }
 
-    return jooq
-      .select(
-        ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS,
-        ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_REFERENCE)
-      .from(ENVIRONMENT_ARTIFACT_VERSIONS)
-      .where(
-        ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(envUid),
-        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artUid),
-        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.eq(veto.version))
-      .fetchOne { (_, reference) ->
-        if (reference != null && !force) {
-          log.warn(
-            "Not vetoing artifact version as it appears to have already been an automated rollback target: " +
-              "deliveryConfig=${deliveryConfig.name}, " +
-              "environment=${veto.targetEnvironment}, " +
-              "artifactVersion=${veto.version}, " +
-              "priorVersionReference=$reference")
-          return@fetchOne false
+    return selectArtifactStatusAndPromotionReferenceInTargetEnvironment(envUid, artUid, veto.version)
```

Comment:
omg that's a long name :) how about `getArtifactDetailsFromEnvironment`?

Reply:
Fixed in 49a07702f